brian2/synapses/synapses.py

```diff
@@ -794,6 +795,41 @@ def __init__(self, source, target=None, model=None, on_pre=None,
                                 'clock_driven',
                                 once=True)
                 continuous.append(single_equation)
+        def recur_check_event_summed(var,eqs):
+            """
+            Recursive Function Used to identify whether a summed variable is 
+            referring to an event-driven variable.
+
+            Parammeters
+            ----------
+            var : str
+                variable that is required for checking
+            eqs : Equations Object
+                Equations object in which we need to check
+
+            Returns
+            ----------
+            val : boolean
+                Return True if a summed variable is referring to an 
+                event-driven variable, otherwise False or continues recursion
+            """
+            val = False
+            for eq in eqs.values():
+                if var in eq.identifiers and eq.varname != var:
+                    if val == True:
+                        break
+                    elif 'summed' in eq.flags:
+                        val = True
+                        break
+                    else:
+                        val = recur_check_event_summed(eq.varname,eqs)
+                    
+            return val
+        # Checking whether a summed variable is referring to an event-driven variable
+        if len(event_vars):
+            for v in event_vars:
+                if recur_check_event_summed(v,model):
+                    print("Event-Driven Variable " + v + " is being summed")
```

Comment:
This could also be slightly simpler, no need for `if len(event_vars)`, the loop will simply do nothing if it is empty.